### PR TITLE
OpenBLAS: move header files, reduce code with muniversal PG 1.1, add missing build dependency, and set NUM_THREADS

### PR DESCRIPTION
**Pull request by MarcusCalhoun-Lopez**
###### Type(s)

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y / Command Line Tools x.y.z

###### Verification 
Have you

- [ ] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Review by michaelld — APPROVED** [review]
The smaller commits read fine; the larger one is a bit difficult to track, but it also looks fine, Let's see how CI/QA/checks do ... tentative approval here

**Comment by MarcusCalhoun-Lopez** [comment]
The checks seemed to have passed.
If it would be helpful, I can try to break down the larger commit into two smaller ones.

**Comment by michaelld** [comment]
No I think the commits are fine as-is. What do you think @NicosPavlov  ?

File: math/OpenBLAS/Portfile [hunk 1]
```diff
@@ -260,61 +187,30 @@ platform darwin 8 {
...
-            set LAPACKE_DIR [exec /bin/ls ${worksrcpath}/lapack-netlib | \
-                                 grep -i lapacke]
-            xinstall -m 755 ${worksrcpath}/lapack-netlib/${LAPACKE_DIR}/include/lapacke_config.h \
-                ${destroot}${prefix}/include/lapacke_config.h
-            xinstall -m 755 ${worksrcpath}/lapack-netlib/${LAPACKE_DIR}/include/lapacke_mangling.h \
-                ${destroot}${prefix}/include/lapacke_mangling.h
-            xinstall -m 755 ${worksrcpath}/lapack-netlib/${LAPACKE_DIR}/include/lapacke_utils.h \
-                ${destroot}${prefix}/include/lapacke_utils.h
-            xinstall -m 755 ${worksrcpath}/lapack-netlib/${LAPACKE_DIR}/include/lapacke.h \
-                ${destroot}${prefix}/include/lapacke.h
```
Comment (NicosPavlov) [inline math/OpenBLAS/Portfile:317]: This part regarding Lapacke headers has been removed. Is it on purpose of has it been forgotten?

**Comment by NicosPavlov** [comment]
It looks good overall, thanks for the changes which make the Portfile far more readable. There is just the part I commented on where I wonder if the headers won't be missing without this.

File: math/OpenBLAS/Portfile (hunk 1 above)
Comment (MarcusCalhoun-Lopez) [reply math/OpenBLAS/Portfile:317]: I believe the Lapacke header files are installed by the port itself in [Makefile.install](https://github.com/xianyi/OpenBLAS/blob/b89fb708caa5a5a32de8f4306c4ff132e0228e9a/Makefile.install#L74).

**Comment by NicosPavlov** [comment]
Yes, I think so. I just checked about the lapacke headers, and it is different (shortcuts to headers in a dedicated include/openblas folder instead of direct copies in include), but I think it is fine.

**Comment by michaelld** [comment]
Approval from both maintainers ... merging! Let's see how these changes play out in the wild ....

**Comment by tux-o-matic** [comment]
Not well, after upgrading the OpenBLAS port:
```shell
% file /opt/local/include/lapacke_utils.h
/opt/local/include/lapacke_utils.h: broken symbolic link to cblas/lapacke_utils.h
```

**Comment by MarcusCalhoun-Lopez** [comment]
I created a [pull request](https://github.com/macports/macports-ports/pull/17662) that attempts to fix the problem.

**Pull request merged**